### PR TITLE
Improve wasm module parsing code. NFC.

### PR DESCRIPTION
**Pull request by sbc100**
Still very minimal but a little more usable now.

### CODE REVIEW

File: tools/webassembly.py [hunk 1]
```diff
@@ -115,29 +108,53 @@ def add_emscripten_metadata(wasm_file):
     f.write(orig[8:])
 
 
+class Module:
+  """Extremely minimal wasm module reader.  Currently only used
+  for parsing the dylink section."""
```
Comment (kripken) [inline tools/webassembly.py:113]: is there a goal to use it for anything else? (I thought the one other use case, for EM_ASMs, went away)

**Review by kripken — APPROVED** [review]

File: tools/webassembly.py (hunk 1 above)
Comment (sbc100) [reply tools/webassembly.py:113]: I think we can/should still look into reading all metadata using this code.   There is a minor difficulty due to active segments but I believe we can find a way around that.

Comment (kripken) [reply tools/webassembly.py:113]: Why? I thought we'd agreed parsing it in binaryen is simpler and faster (now that we skip functions and names).

Comment (sbc100) [reply tools/webassembly.py:113]: I guess I hadn't fully understood that conclusion.   I'm on board with the idea of keeping binaryen around for metadata parsing if it really is faster.   But the idea of completely removing the brittle ABI between emscripten and binaryen is still somewhat alluring to me.    I agree its not worth investing in right now given all our other priorities though.

Comment (sbc100) [reply tools/webassembly.py:113]: Ah I forgot, I'm also planning to use this code to read the imports from shared libraries as part of #13830

Comment (kripken) [reply tools/webassembly.py:113]: Oh, that's an interesting use case. Let's discuss further there.

**Pull request merged**